src/EditorFeatures/Core/Implementation/IntelliSense/AsyncCompletion/CommitManager.cs

```diff
@@ -268,5 +287,42 @@ internal static bool SendEnterThroughToEditor(CompletionRules rules, RoslynCompl
                     return item.GetEntireDisplayText() == textTypedSoFar;
             }
         }
+
+        internal static string AdjustForVirtualSpace(TextChange textChange, ITextView textView, IEditorOperationsFactoryService editorOperationsFactoryService)
```

Comment:
Can we just make this the editor's problem? It seems odd we're having to do this same workaround in the new API.

Reply:
It seems we always can pass DefaultMinimalChange to the Editor. Then, we can remove this code back to the old completion and do not use it in the new one.

---
In reply to: [239972184](https://github.com/dotnet/roslyn/pull/31596#discussion_r239972184) [](ancestors = 239972184)

Reply:
We also need adjustedNewText below. Need some more time to consider how to solve this.

---
In reply to: [239985219](https://github.com/dotnet/roslyn/pull/31596#discussion_r239985219) [](ancestors = 239985219,239972184)

Reply:
Will consider moving editor operations out of Roslyn later: [https://github.com/dotnet/roslyn/issues/31670](https://github.com/dotnet/roslyn/issues/31670) to track